sw/airborne/modules/multi/follow.c

```diff
@@ -47,32 +45,44 @@
 #define FOLLOW_OFFSET_Z 0.0
 #endif
 
-void follow_init(void)
-{
+#ifndef FOLLOW_AC_ID
+#error "Please define FOLLOW_AC_ID"
+#endif
 
-}
+#ifndef FOLLOW_WAYPOINT_ID
+#error "Please define FOLLOW_WAYPOINT_ID"
+#endif
 
-void follow_change_wp(unsigned char *buffer)
+void follow_init(void){}
+
+/*
+ * follow_wp(void)
+ * updates the FOLLOW_WAYPOINT_ID to a fixed offset from the last received location
+ * of other aircraft with id FOLLOW_AC_ID
+ */
+void follow_wp(void)
 {
-  struct EcefCoor_i new_pos;
-  struct EnuCoor_i enu;
-  new_pos.x = DL_REMOTE_GPS_ecef_x(buffer);
-  new_pos.y = DL_REMOTE_GPS_ecef_y(buffer);
-  new_pos.z = DL_REMOTE_GPS_ecef_z(buffer);
+  struct ac_info_ * ac = get_ac_info(FOLLOW_AC_ID);
+
+  // todo replace with stateGet function when they are working for utm pos
+  struct UtmCoor_f my_pos = utm_float_from_gps(&gps, 0);
+  my_pos.alt = gps.hmsl / 1000.;
 
-  // Translate to ENU
-  enu_of_ecef_point_i(&enu, &state.ned_origin_i, &new_pos);
-  INT32_VECT3_SCALE_2(enu, enu, INT32_POS_OF_CM_NUM, INT32_POS_OF_CM_DEN);
+  // get the distance to the other vehicle to be followed
+  struct EnuCoor_f enu = stateGetPositionEnu_f();
+  enu.x += ac->utm.east/100. - my_pos.east;
```

Comment:
still don't understand what this is about...
why not simply use the position of the other vehicle and then add the FOLLOW_OFFSET as before?